### PR TITLE
Fix: Temporary solution to translate report message

### PR DESCRIPTION
**Pull request by habemuscode**
**Ticket**

https://oat-sa.atlassian.net/browse/ADF-438

**Related PR's**

- https://github.com/oat-sa/extension-tao-itemqti/pull/1773
- https://github.com/oat-sa/extension-tao-task-queue/pull/133
- https://github.com/oat-sa/tao-core/pull/2950
- https://github.com/oat-sa/generis/pull/900

### CODE REVIEW

File: src/taskQueue/taskQueueModel.js [hunk 1]
```diff
@@ -71,6 +72,18 @@ function hasSameState(task1, task2) {
     return false;
 }
 
+function workaroundMessages(taskData) {
```
Comment (gabrielfs7) [inline src/taskQueue/taskQueueModel.js:75]: could we maybe rename it `translateMessages`?

Comment (gabrielfs7) [reply src/taskQueue/taskQueueModel.js:75]:
```suggestion
function translateReportMessages(reportChildren) {
```

File: src/taskQueue/taskQueueModel.js [hunk 2]
```diff
@@ -71,6 +72,18 @@ function hasSameState(task1, task2) {
...
+    return taskData.map(item => {
+        if (item.interpolationMessage) {
+            item.message = __(item.interpolationMessage, ...item.interpolationData);
```
Comment (gabrielfs7) [inline src/taskQueue/taskQueueModel.js:78]: Just a heads up that `interpolationData` can be an empty array sometimes.

Comment (404passionFruit) [reply src/taskQueue/taskQueueModel.js:78]: Should be fine with empty arrays as long as it's an array and not undefined or null

**Review by 404passionFruit — COMMENTED** [review]
Agree, the method should be renamed

**Review by gabrielfs7 — APPROVED** [review]
LGTM 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: package.json [hunk 3]
```diff
@@ -87,7 +87,8 @@
         "rollup-plugin-node-resolve": "^5.0.0",
         "select2": "^3.5.1",
         "text": "github:requirejs/text#2.0.16",
-        "tooltip.js": "^1.3.1"
+        "tooltip.js": "^1.3.1",
+        "eve": "git+https://github.com/adobe-webplatform/eve.git#eef80ed"
```
Comment (bziondik) [inline package.json:91]: Please, add `package.json`

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on playground (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Comment by 404passionFruit** [comment]
There are timeout issues on CI tests looking for a way to fix them

File: src/taskQueue/taskQueueModel.js [hunk 4]
```diff
@@ -145,6 +158,8 @@ export default function taskQueueModel(config) {
             }
 
             status = request(config.url.get, { taskId: taskId }, 'GET', {}, true).then(function(taskData) {
+                // Workaround for translations
+                taskData.report.children = translateReportMessages(taskData.report.children);
```
Comment (gabrielfs7) [inline src/taskQueue/taskQueueModel.js:162]: @habemuscode Missing some validation here?

```
if (typeof taskData.report !== 'undefined' && typeof taskData.report.children !== 'undefined') {
  taskData.report.children = translateReportMessages(taskData.report.children); //check taskData 
}
```

**Pull request merged**